azure/Guidelines.md

```diff
@@ -88,80 +86,53 @@ All Azure APIs **MUST** use explicit versioning. The Microsoft REST API guidelin
 
-If a new property is made required in the request body, clients will have no way to set this and the request will fail.
+Azure services **MUST** update the version number of their API whenever there is a change to the API, no matter how small.  Customers will "lock the API version" so that their code does not fail when the service introduces new features.  They rely on the fact that an API version is a contract with the services that will never change.
 
-#### Property name has changed
+A breaking change is any change in the API that may cause client or service code making the API call to fail. Obvious examples of such a change are the removal of an endpoint, adding or removing a required field or changing the format of the body (from XML to JSON for example). Even though we recommend clients ignore new fields, there are many libraries and clients that fail when new fields are introduced.
 
-Note that this is implied by the requirement that adding/and removing properties are breaking changes, but in some ways worse, since it leads to the possibility of reusing a property name. Even with an API version change, this change is discouraged because it creates documentation and cognitive challenges.
+There are three reasons why a service may issue a breaking change:
 
-#### Property type has changed
+1. To remove a security vulnerability.
+2. To comply with regulatory requirements.
+3. To deprecate a feature of the service.
 
-Property `foo was a boolean in v1 but is changed to a string. A client using the existing API version tries to set it as a boolean, but the service will fail since its now expecting a string. So, the API version must be updated.
+In each case, prior approval of the Azure REST API review board is required.  In the case of deprecation, follow the API deprecation policy (below).  If the service is using SemVer for versioning, breaking changes constitute a major version change.
 
-#### Property default value has changed
+#### Examples of breaking changes
 
-If a property is optional and the service provides a default value, changing that default requires an updated API version.
+At a high level, any change to the contract of an API constitutes a breaking change. The following is a non-exhaustive list of breaking changes:
 
-#### Allowed values for an enum have changed
-Enum “foo” had allowed values as “val1” and “val2” in v1 of API. Now, the values accepted by the service are “val1”, “val2” and “val3”.  The client will fail to de-serialize if “val3” comes back in the response.
+* An existing property is removed.
+* A new property is added to an existing response.
+* A new required property is added to an existing request.
+* A property name is changed (including case changes).
+* A property type is changed.
+* The default value of a property is changed.
+* The allowed values for an enum is changed.
+* An API is removed.
+* The behavior of an existing API is changed.
+* The error contract has changed.
```

Comment:
Does this include the string contents of an error message? Does changing an error message count as a breaking change? For Cosmos, these are frequently iterated on to provide better debugging information to the user. I'm fairly sure we don't consider these changes breaking.